keygen/src/keygen.rs

```diff
@@ -353,6 +376,23 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                 .collect::<Vec<_>>();
             thread::park();
         }
+        ("verify", Some(matches)) => {
+            let keypair = get_keypair_from_matches(matches)?;
+            let test_data = "test".as_bytes();
+            let signature = Signature::new(&keypair.sign(&test_data).to_bytes());
+            let pubkey = if matches.is_present("pubkey") {
```

Comment:
I think the `pubkey` should be required.  Using the one generated from the input keypair (assuming it resolves to a keypair) will always pass `signature.verify()` below.